recipes-devtools/deepstream/deepstream-6.1_6.1.0-1.bb

```diff
@@ -82,6 +89,32 @@ do_install() {
     cp -R --preserve=mode,timestamps ${S}${DEEPSTREAM_PATH}/sources/includes/* ${D}${includedir}/
 
     cp -R --preserve=mode,timestamps ${S}${DEEPSTREAM_PATH}/sources/ ${D}${DEEPSTREAM_PATH}/
+
+    # XXX---
+    # Some of the binaries are not using the right SONAME
+    # in its DT_NEEDED ELF header, so we have to rewrite it to prevent
+    # a broken runtime dependency.
+    patchelf --replace-needed libyaml-cpp.so.0.6 libyaml-cpp.so.0.7 ${D}${bindir}/deepstream-transfer-learning-app
+    patchelf --replace-needed libyaml-cpp.so.0.6 libyaml-cpp.so.0.7 ${D}${bindir}/deepstream-test5-app
+    patchelf --replace-needed libyaml-cpp.so.0.6 libyaml-cpp.so.0.7 ${D}${bindir}/deepstream-app
+    patchelf --replace-needed libyaml-cpp.so.0.6 libyaml-cpp.so.0.7 ${D}${bindir}/deepstream-3d-depth-camera
+    patchelf --replace-needed libyaml-cpp.so.0.6 libyaml-cpp.so.0.7 ${D}${libdir}/gstreamer-1.0/deepstream/libnvdsgst_multistream.so
+    patchelf --replace-needed libyaml-cpp.so.0.6 libyaml-cpp.so.0.7 ${D}${libdir}/gstreamer-1.0/deepstream/libnvdsgst_infer.so
+    patchelf --replace-needed libyaml-cpp.so.0.6 libyaml-cpp.so.0.7 ${D}${libdir}/gstreamer-1.0/deepstream/libnvdsgst_dsanalytics.so
+    patchelf --replace-needed libyaml-cpp.so.0.6 libyaml-cpp.so.0.7 ${D}${libdir}/gstreamer-1.0/deepstream/libnvdsgst_3dfilter.so
+    patchelf --replace-needed libyaml-cpp.so.0.6 libyaml-cpp.so.0.7 ${D}${DEEPSTREAM_PATH}/lib/libpostprocess_impl.so
+    patchelf --replace-needed libyaml-cpp.so.0.6 libyaml-cpp.so.0.7 ${D}${DEEPSTREAM_PATH}/lib/libnvds_yml_parser.so
+    patchelf --replace-needed libyaml-cpp.so.0.6 libyaml-cpp.so.0.7 ${D}${DEEPSTREAM_PATH}/lib/libnvds_3d_depth2point_datafilter.so
+    patchelf --replace-needed libyaml-cpp.so.0.6 libyaml-cpp.so.0.7 ${D}${DEEPSTREAM_PATH}/lib/libnvds_3d_depth_datasource.so
+    patchelf --replace-needed libyaml-cpp.so.0.6 libyaml-cpp.so.0.7 ${D}${DEEPSTREAM_PATH}/lib/libnvds_3d_gl_datarender.so
+    patchelf --replace-needed libyaml-cpp.so.0.6 libyaml-cpp.so.0.7 ${D}${DEEPSTREAM_PATH}/lib/libnvds_msgconv.so.1.0.0
+    patchelf --replace-needed libyaml-cpp.so.0.6 libyaml-cpp.so.0.7 ${D}${DEEPSTREAM_PATH}/lib/libnvds_nvmultiobjecttracker.so
+    patchelf --replace-needed libdns_sd.so.1.0.0 libdns_sd.so.1 ${D}${DEEPSTREAM_PATH}/lib/libnvds_nvmultiobjecttracker.so
+    patchelf --replace-needed libdns_sd.so.1.0.0 libdns_sd.so.1 ${D}${DEEPSTREAM_PATH}/lib/libnvds_nmos.so
```

Comment:
These for `libdns_sd` look strange.  Probably OK, as I can't think of a better workaround.